src/main/java/lotto/domain/LottoResultCalculator.java

```diff
@@ -0,0 +1,64 @@
+package lotto.domain;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lotto.dto.LottoResult;
+import lotto.dto.RankCount;
+import lotto.dto.RankCounts;
+import lotto.dto.WinningLottoNumbers;
+import lotto.util.enums.Rank;
+
+public class LottoResultCalculator {
+
+    public LottoResult calculateResult(Lottos lottos, WinningLottoNumbers winningLottoNumbers, int purchaseAmount) {
+        Map<Rank, Integer> rankCountMap = initializeRankCountMap();
+        List<Integer> winningNumbers = winningLottoNumbers.getWinningNumbers();
+        int bonusNumber = winningLottoNumbers.getBonusNumber();
+        updateRankCountMap(rankCountMap, lottos, winningNumbers, bonusNumber);
+
+        RankCounts rankCounts = convertToRankCounts(rankCountMap, purchaseAmount);
+
+        return new LottoResult(rankCounts, purchaseAmount);
+    }
+
+    private Map<Rank, Integer> initializeRankCountMap() {
+        return Arrays.stream(Rank.values())
+                .collect(Collectors.toMap(rank -> rank, rank -> 0, (a, b) -> b, () -> new EnumMap<>(Rank.class)));
+    }
+
+    private void updateRankCountMap(Map<Rank, Integer> rankCountMap, Lottos lottos, List<Integer> winningNumbers,
+                                    int bonusNumber) {
+        lottos.getLottos().stream()
+                .map(lotto -> lotto.calculateRank(winningNumbers, bonusNumber))
+                .forEach(rank -> rankCountMap.put(rank, rankCountMap.get(rank) + 1));
+    }
```

Comment:
개인적으로 메서드의 파라미터가 많아질 경우 다음과 같은 방법을 사용하는 것이 가독성에 더 좋다고 생각해요!
```suggestion
    private void updateRankCountMap(
        Map<Rank, Integer> rankCountMap,
        Lottos lottos,
        List<Integer> winningNumbers,
        int bonusNumber
    ) {
        lottos.getLottos().stream()
                .map(lotto -> lotto.calculateRank(winningNumbers, bonusNumber))
                .forEach(rank -> rankCountMap.put(rank, rankCountMap.get(rank) + 1));
    }
```